src/main/java/hudson/plugins/git/util/GitUtils.java

```diff
@@ -272,6 +272,15 @@ private static void addEnvironmentContributingActionsValues(EnvVars env, Abstrac
                 }
             }
         }
+        
+        // Use the default parameter values (if any) instead of the ones from the last build
```

Comment:
If job will have sha1 default value (or any other sensitive), then it will screw the whole gitscm. Other plugins in such cases filtering them. Magic not obvious parameter is bad behaviour.